script/install.sh

```diff
@@ -673,67 +729,120 @@ write_env_vars() {
             local dc=gdc
             local dmd=gdmd
             ;;
+        dub*)
+            local libpath=
+            local dc=
+            local dmd=
+            ;;
     esac
 
     logV "Writing environment variables to $ROOT/$1/activate"
     cat > "$ROOT/$1/activate" <<EOF
 deactivate() {
     export PATH="\$_OLD_D_PATH"
+EOF
+    if [ -n "$libpath" ] ; then
+    cat >> "$ROOT/$1/activate" <<EOF
     export LIBRARY_PATH="\$_OLD_D_LIBRARY_PATH"
     export LD_LIBRARY_PATH="\$_OLD_D_LD_LIBRARY_PATH"
-    export PS1="\$_OLD_D_PS1"
-
-    unset _OLD_D_PATH
     unset _OLD_D_LIBRARY_PATH
     unset _OLD_D_LD_LIBRARY_PATH
-    unset _OLD_D_PS1
+EOF
+    fi
+    if [ -n "$dmd" ] ; then
+    cat >> "$ROOT/$1/activate" <<EOF
     unset DMD
     unset DC
+EOF
+    fi
+
+    cat >> "$ROOT/$1/activate" <<EOF
+    export PS1="\$_OLD_D_PS1"
+    unset _OLD_D_PATH
+    unset _OLD_D_PS1
     unset -f deactivate
 }
 
 _OLD_D_PATH="\${PATH:-}"
+EOF
+
+    if [ -n "$libpath" ] ; then
+    cat >> "$ROOT/$1/activate" <<EOF
 _OLD_D_LIBRARY_PATH="\${LIBRARY_PATH:-}"
 _OLD_D_LD_LIBRARY_PATH="\${LD_LIBRARY_PATH:-}"
-_OLD_D_PS1="\${PS1:-}"
-
-export PATH="${DUB_BIN_PATH}${DUB_BIN_PATH:+:}$ROOT/$1/$binpath\${PATH:+:}\${PATH:-}"
 export LIBRARY_PATH="$ROOT/$1/$libpath\${LIBRARY_PATH:+:}\${LIBRARY_PATH:-}"
 export LD_LIBRARY_PATH="$ROOT/$1/$libpath\${LD_LIBRARY_PATH:+:}\${LD_LIBRARY_PATH:-}"
+EOF
+    fi
+
+    cat >> "$ROOT/$1/activate" <<EOF
+_OLD_D_PATH="\${PATH:-}"
+_OLD_D_PS1="\${PS1:-}"
+export PS1="($1)\${PS1:-}"
+export PATH="${DUB_BIN_PATH}${DUB_BIN_PATH:+:}$ROOT/$1/$binpath\${PATH:+:}\${PATH:-}"
+EOF
+
+    if [ -n "$dmd" ] ; then
+    cat >> "$ROOT/$1/activate" <<EOF
 export DMD=$dmd
 export DC=$dc
-export PS1="($1)\${PS1:-}"
 EOF
+    fi
 
     logV "Writing environment variables to $ROOT/$1/activate.fish"
     cat > "$ROOT/$1/activate.fish" <<EOF
 function deactivate
     set -gx PATH \$_OLD_D_PATH
+EOF
+
+if [ -n "$libpath" ] ; then
+    cat >> "$ROOT/$1/activate.fish" <<EOF
     set -gx LIBRARY_PATH \$_OLD_D_LIBRARY_PATH
     set -gx LD_LIBRARY_PATH \$_OLD_D_LD_LIBRARY_PATH
+    set -e _OLD_D_LIBRARY_PATH
+    set -e _OLD_D_LD_LIBRARY_PATH
+EOF
+fi
 
+cat >> "$ROOT/$1/activate.fish" <<EOF
     functions -e fish_prompt
     functions -c _old_d_fish_prompt fish_prompt
     functions -e _old_d_fish_prompt
 
     set -e _OLD_D_PATH
-    set -e _OLD_D_LIBRARY_PATH
-    set -e _OLD_D_LD_LIBRARY_PATH
+EOF
+
+if [ -n "$dmd" ] ; then
+    cat >> "$ROOT/$1/activate.fish" <<EOF
     set -e DMD
     set -e DC
+EOF
+fi
+cat >> "$ROOT/$1/activate.fish" <<EOF
```

Comment:
Might be less messy to have one scope per file instead of all these appends.

https://github.com/koalaman/shellcheck/wiki/SC2129